### PR TITLE
3 create value set list builder

### PR DESCRIPTION
**Pull request by Frontman50**
builds valueSets.default.json based on the folder given. Has option for automatic language detection --lang_detection

### CODE REVIEW

File: generate_default_value_set_list.py [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    )
+    logging.getLogger("deepl").disabled = True
+
+
+if __name__ == "__main__":
+    args = configure_args_parser().parse_args()
+    configure_logging(args.log_level)
+
+    deepl_engine = None
+    if args.deepl_api_key != "no_key":
```
Comment (juliangruendner) [inline generate_default_value_set_list.py:45]: @Frontman50 i would do this differently - check if --lang_detection == true and if so and no api key return with an error -

File: generate_default_value_set_list.py [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+                source_lang = "de"
+                if args.lang_detection:
+                    with open(os.path.join(folder_path, f), "r", encoding="utf-8") as json_file:
+                        json_data = json.load(json_file)
+                        value_set_sample = json_data["expansion"]["contains"][0]["display"]
+                    result = deepl_engine.translate_text(text=value_set_sample, target_lang="DE")
+                    logger.info(value_set_sample + " -> source: " + result.detected_source_lang)
+                    source_lang = result.detected_source_lang
+
+                files.append({
```
Comment (juliangruendner) [inline generate_default_value_set_list.py:65]: url should not be a file path - should be the url from the value set object instead 
as in - read in the json and get the "url" attribute from the value set json

File: generate_default_value_set_list.py [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+        files = []
+        for f in os.listdir(folder_path):
+            if os.path.isfile(os.path.join(folder_path, f)) and f.endswith(".json"):
+
+                source_lang = "de"
+                if args.lang_detection:
+                    with open(os.path.join(folder_path, f), "r", encoding="utf-8") as json_file:
+                        json_data = json.load(json_file)
+                        value_set_sample = json_data["expansion"]["contains"][0]["display"]
+                    result = deepl_engine.translate_text(text=value_set_sample, target_lang="DE")
```
Comment (juliangruendner) [inline generate_default_value_set_list.py:61]: does this always work?
if the target_lang = DE does it still detect a DE source_lang?

**Review by juliangruendner — CHANGES_REQUESTED** [review]

File: generate_default_value_set_list.py (hunk 3 above)
Comment (Frontman50) [reply generate_default_value_set_list.py:66]: It does work quite consistently. See below

2024-10-25 10:16:11,303 - INFO - (Teilweiser) Verlust der oberen Extremität, beidseitig -> source: DE
2024-10-25 10:16:11,303 - INFO - added http://fhir.de/ValueSet/bfarm/icd-10-gm ->DE
2024-10-25 10:16:11,394 - INFO - (Analgo-)Sedierung -> source: DE
2024-10-25 10:16:11,394 - INFO - added http://fhir.de/ValueSet/bfarm/ops ->DE
2024-10-25 10:16:11,457 - INFO - Allgemeine Chirurgie -> source: DE
2024-10-25 10:16:11,457 - INFO - added http://fhir.de/ValueSet/dkgev/Fachabteilungsschluessel ->DE

**Pull request merged**